cmd/minishift/cmd/start_preflight_test.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"testing"
+
+	configCmd "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/spf13/viper"
+)
+
+func TestIsoUrl(t *testing.T) {
+	var isoURLCheck = []struct {
+		in  string
+		out bool
+	}{
+		{"http://github.com/minishift/minishift-centos/minishift-centos.iso", true},
+		{"https://github.com/minishift//minishift/minishift-b2d.iso", true},
+		{"file:///home/foo/bar/bar.iso", true},
+		{"file://home/bar/baz.iso", false},
+		{"file://minishift/hoxier/foo", false},
+		{"/home/joey/chandler/iso.iso", false},
+		{"file:///home/homie/foo", false},
+		{"ram/rahim/anthony.iso", false},
+		{"gopher/minishift/b2d", false},
+		{"file://c:/minishift/gopher.iso", true},
```

Comment:
You code is OS specific, your tests will have to deal with that as well. See also https://dave.cheney.net/2013/10/12/how-to-use-conditional-compilation-with-the-go-build-tool

Reply:
@anjannath add a test in a file `start_preflight_test_windows.go` which test for the `file://c:/` situation.

Reply:
the tests now will test for both OS, and the regular expression is exercised already, only the `os.stat` was failing.